### PR TITLE
chore(developer): remove obsolete 'Allow Multiple Instances' and 'Use Legacy Compiler' options

### LINKED ISSUE #9925: chore(developer): remove obsolete settings 'Allow multiple instances', 'Use legacy compiler'
* Allow multiple instances should always be true
* Use legacy compiler is now irrelevant for 17.0

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9925.

Allow Multiple Instances is now always true.
Use Legacy Compiler did nothing (there is no legacy compiler!).

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/tike/main/KeymanDeveloperUtils.pas [hunk 1]
```diff
@@ -192,54 +188,6 @@ function GetFolderPath(csidl: Integer): string;
...
-      else FMultiInstance := False;
-  finally
-    Free;
-  end;
-
-  if FMultiInstance then
-  begin
-    Result := False;
-    Exit;
-  end;
```
Comment (mcdurdin) [inline developer/src/tike/main/KeymanDeveloperUtils.pas:214]: always multi instance now, so this func always does nothing

File: common/windows/delphi/general/RegistryKeys.pas [hunk 2]
```diff
@@ -103,20 +103,16 @@ interface
 
   SRegValue_Engine_OEMProductPath = 'oem product path';
 
-//  SRegValue_UnknownLayoutID         = 'unknown layout id';                          // LM
```
Comment (mcdurdin) [inline common/windows/delphi/general/RegistryKeys.pas:106]: removing lots of commented out entries as a side-chore

File: developer/src/tike/main/UfrmMain.pas [hunk 3]
```diff
@@ -761,31 +760,6 @@ procedure TfrmKeymanDeveloper.WMUserInputLangChange(var Message: TMessage);
     PostMessage(FChildWindows[i].Handle, WM_USER_INPUTLANGCHANGE, Message.wParam, Message.LParam);
 end;
 
-procedure TfrmKeymanDeveloper.WMUserLoadRegFiles(var Message: TMessage);
```
Comment (mcdurdin) [inline developer/src/tike/main/UfrmMain.pas:764]: Only used by TikeActive call to load files into existing instance... so no longer used.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.205-alpha](https://keyman.com/downloads/releases/alpha/17.0.205)

**Pull request merged**